src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +189,111 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             }
         }));
     }
+
+    private async Task WatchDashboardLogsAsync()
+    {
+        async Task<string?> GetDashboardResourceIdentifier()
+        {
+            string? dashboardResourceId = null;
+
+            try
+            {
+                // The dashboard resource isn't immediately available so watch for it.
+                // Wait for dashboard to startup and be reported before giving up.
+                using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30));
+
+                await foreach (var notification in resourceNotificationService.WatchAsync(cts.Token))
+                {
+                    // We want to capture the replica of the dashboard resource. This will allow us to get the logs for the dashboard.
+                    // When the resource id is not the same as the resource name, it's the replica.
+                    if (notification.Resource.Name == KnownResourceNames.AspireDashboard && notification.ResourceId != notification.Resource.Name)
+                    {
+                        dashboardResourceId = notification.ResourceId;
+                        break;
+                    }
+                }
+            }
+            catch (OperationCanceledException)
+            {
+            }
+
+            return dashboardResourceId;
+        }
+
+        if (await GetDashboardResourceIdentifier().ConfigureAwait(false) is not string dashboardResourceId)
+        {
+            // The only way this can happen is if DCP is unhealthy, so log a warning.
+            distributedApplicationLogger.LogWarning("Failed to get the dashboard logs.");
+            return;
+        }
+
+        var loggerCache = new Dictionary<string, ILogger>();
```

Comment:
Be explicit about logger name match?
```suggestion
        var loggerCache = new Dictionary<string, ILogger>(StringComparer.Ordinal);
```